### ISSUE TITLE
Rule Change: option to turn off max lines or statements for constructor functions

### ISSUE THREAD

**Issue by gusborg88** [eslint/eslint#19294]
### What rule do you want to change?

max-lines-per-function, max-statements

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
function ConstructorFunction() {
  this.myFavNumber = 400;
  this.presidentWhoJustDied = 'jimmy cahtuh';
  this.method = () => {
    console.log(this.myFavNumber);
  }
  /*many more lines and statements*/
}
```

### What does the rule currently do for this code?

Shows an error if a function has more than 10 statements or more than 50 lines

### What will the rule do after it's changed?

Add an option like ignoreObjectConstructors:boolean to turn off the rule for object constructors

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Classes don't have a limit to lines or statements (I think), and object constructors are like classes.

**Comment by nzakas** [comment]
Thanks for the suggestion. This is a fairly narrow exception, especially because constructor functions of this nature aren't really used much anymore. As such, it's not a good candidate to update core rules for this use case.
